### PR TITLE
Build JNI on Windows

### PR DESCRIPTION
**Pull request by sstamenova**
This change makes JNI buildable on Windows. It also cleans up the logic for determining when to build JNI and adds the necessary dependency to MainUtils

### CODE REVIEW

**Comment by caoimhinuibrian** [comment]
@jenkins-droid test this please

File: src/Runtime/jni/jnilog.c [hunk 1]
```diff
@@ -86,9 +120,10 @@ static FILE *get_log_file_by_name(char *name) {
   else if (!strcmp(name, "stderr"))
     fp = stderr;
   else {
-    char tname[strlen(name) + 32];
-    snprintf(tname, strlen(name) + 32, "%s.%lx", name, pthread_self());
+    char *tname = malloc(strlen(name) + 32);
+    snprintf(tname, strlen(name) + 32, "%s.%lx", name, get_threadid());
     fp = fopen(tname, "w");
+    free(tname);
   }
```
Comment (gongsu832) [inline src/Runtime/jni/jnilog.c:127]: Check tname for NULL:
```
char *tname = malloc(strlen(name) + 32);
if (tname) {
  snprintf(tname, strlen(name) + 32, "%s.%lx", name, get_threadid());
  fp = fopen(tname, "w");
  free(tname);
}
```

Comment (AlexandreEichenberger) [reply src/Runtime/jni/jnilog.c:129]: tx

**Comment by caoimhinuibrian** [comment]
@jenkins-droid test this please

**Comment by sstamenova** [comment]
@gongsu832 : Let me know if there's anything else in the change that you'd like addressed.

**Comment by gongsu832** [comment]
LGTM @jenkins-droid test this please

**Comment by sstamenova** [comment]
@gongsu832 : Is there anything blocking this change from going in? Can we commit it?

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Comment by AlexandreEichenberger** [comment]
@gongsu832 Are you ok with this PR? @sstamenova wants to commit this one before looking into fixing #744 for windows. Since this is JNI, you should really be the one giving the final ok on our side, thanks @gongsu832

**Comment by gongsu832** [comment]
@AlexandreEichenberger Yes it's OK to merge. Thanks.

**Review by caoimhinuibrian — APPROVED** [review]
LGTM

**Pull request merged**